pkg/kubectl/cmd/create_pvc.go

```diff
@@ -0,0 +1,210 @@
+
+		# Create a persistent volume claim named "myclaim" with multiple access modes and "1Gi" capacity.
+		kubectl create persistentvolumeclaim myclaim --access-mode=RWO,RWX --capacity=1Gi`))
+)
+
+type CreatePVCOptions struct {
+	Name         string
+	AccessModes  []api.PersistentVolumeAccessMode
+	Capacity     *resource.Quantity
+	StorageClass *string
+	VolumeName   string
+
+	DryRun       bool
+	OutputFormat string
+	Namespace    string
+	Client       internalversionstorage.PersistentVolumeClaimInterface
+	Mapper       meta.RESTMapper
+	Out          io.Writer
+	PrintObject  func(obj runtime.Object) error
+}
+
+// NewCmdCreatePersistentVolumeClaim is a command to ease creating PersistentVolumeClaims.
+func NewCmdCreatePersistentVolumeClaim(f cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	c := &CreatePVCOptions{
+		Out: cmdOut,
+	}
+	cmd := &cobra.Command{
+		Use:     "persistentvolumeclaim NAME --access-mode=accessmode --capacity=capacity [--storage-class=storageclass] [--volume-name=volumename] [--dry-run]",
+		Aliases: []string{"pvc"},
+		Short:   pvcLong,
+		Long:    pvcLong,
+		Example: pvcExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(c.Complete(f, cmd, args))
+			cmdutil.CheckErr(c.Validate())
+			cmdutil.CheckErr(c.RunCreatePVC())
+		},
+	}
+	cmdutil.AddApplyAnnotationFlags(cmd)
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddPrinterFlags(cmd)
+	cmdutil.AddDryRunFlag(cmd)
+
+	cmd.Flags().StringSlice("access-mode", []string{}, "AccessMode contains the desired access modes the volume should have.")
+	cmd.Flags().String("capacity", "", "Capacity represents the minimum storage capacity required.")
+	cmd.Flags().String("storage-class", "", "Name of the StorageClass required by the claim.")
+	cmd.Flags().StringVar(&c.VolumeName, "volume-name", "", "VolumeName is the binding reference to the PersistentVolume backing this claim.")
+
+	return cmd
+}
+
+func (c *CreatePVCOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	name, err := NameFromCommandArgs(cmd, args)
+	if err != nil {
+		return err
+	}
+	c.Name = name
+
+	// Transform access mode abbreviation to complete access mode.
+	accessModes := []api.PersistentVolumeAccessMode{}
+	modes := cmdutil.GetFlagStringSlice(cmd, "access-mode")
+	for _, m := range modes {
+		var accessMode api.PersistentVolumeAccessMode
+		switch m {
+		case "RWX":
+			accessMode = api.ReadWriteMany
+		case "RWO":
+			accessMode = api.ReadWriteOnce
+		case "ROX":
+			accessMode = api.ReadOnlyMany
+		default:
+			accessMode = api.PersistentVolumeAccessMode(m)
+		}
+		accessModes = append(accessModes, accessMode)
+	}
+	c.AccessModes = accessModes
+
+	// If user does not provide storage-class option, then StorageClass will be set as nil.
+	// Otherwise, StorageClass will be the value set by user, including empty string.
+	storageClassFlag := cmd.Flags().Lookup("storage-class")
+	if storageClassFlag.Changed {
```

Comment:
Still open on `storageClass`. I'd prefer [above discussion](https://github.com/kubernetes/kubernetes/pull/52006#r137437564).

@xingzhou If using this checking, how about changing to 

```
if cmd.Flags().Changed("storage-class")
```

Reply:
This is what I have tried in my local env:
1. `kubectl create persistentvolumeclaim abc --storage-class="" --capacity=1Gi --access-mode=RWO`, after this command, using `kubectl get pvc abc -o yaml`, the `storageClassName` field is set as `storageClassName:""`

2. `kubectl create persistentvolumeclaim abc  --capacity=1Gi --access-mode=RWO`, after this command, using `kubectl get pvc abc -o yaml`, the `storageClassName` field is set as `storageClassName:standard`, note that it's using the default storage class

3.`kubectl create persistentvolumeclaim abc --storage-clas=abc --capacity=1Gi --access-mode=RWO`, after this command, using `kubectl get pvc abc -o yaml`, the `storageClassName` field is set as `storageClassName: abc`

I tried the similar things using yaml file, the result is the same as using this command code. And I think to distinguish `nil`, empty string and other string makes sense for the following provisioning process, so I kept this check. 

But I'll try `if cmd.Flags().Changed("storage-class")` as it definitely will simplify the code. Will let you know the result later.

Reply:
Have tried `cmd.Flags().Changed("storage-class")` and it worked, I've updated the patch and PTAL, thanks!

Reply:
@xingzhou I submitted a PR #52075 on the `storageClass`. PTAL. I think this pr is related to this issue (assigning empty string as `storageClass` name).

Also please help verify if #52075 resolves this. Thanks so much.